src/Tribe/Ticket_Object.php

```diff
@@ -507,11 +507,20 @@ public function is_in_stock() {
 				return true;
 			}
 
-			$remaining = $this->inventory();
+			/* @var Tribe__Cache $cache */
+			$cache = tribe( 'cache' );
+			$key   = __METHOD__ . '-' . $this->ID;
 
-			$is_unlimited = $remaining === -1;
+			if ( isset( $cache[ $key ] ) ) {
+				return $cache[ $key ];
+			}
+
+			$remaining    = $this->inventory();
+			$is_unlimited = $remaining === - 1;
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: e23a36edce53cfad224a513761989588</sub>

Reply:
```suggestion
			$is_unlimited = - 1 === $remaining;
```